flexget/plugins/metainfo/nfo_lookup.py

```diff
@@ -0,0 +1,256 @@
+                    return
+
+            # Populate the fields from the information in the .nfo file Note
+            # that at this point `nfo_filename` has the name of an existing
+            # '.nfo' file
+            self.lookup(entry, nfo_filename)
+
+    def lookup(self, entry, nfo_filename):
+        # If there is already data from a previous parse then we don't need to
+        # do anything
+        if entry.get('nfo_id') is not None:
+            log.warning("Entry %s was already parsed by nfo_lookup and it "
+                        "will be skipped. ", entry.get('title'))
+            return
+
+        # nfo_filename Should not be None at this point
+        assert nfo_filename is not None
+
+        # Get all values we can from the nfo file. If the nfo file can't be
+        # parsed then a warning is logged and we return without changing the
+        # entry
+        try:
+            nfo_reader = NfoReader(nfo_filename)
+            fields = nfo_reader.get_fields_from_nfo_file()
+        except BadXmlFile:
+            log.warning("Invalid '.nfo' file for entry %s", entry.get('title'))
+            return
+
+        entry.update(fields)
+
+        # If the IMDB id was found in the nfo file, set the imdb_id field of
+        # the entry. This will help the imdb_lookup plugin to get the correct
+        # data if it is also used.
+        if 'nfo_id' in fields:
+            entry.update({u'imdb_id': fields['nfo_id']})
+
+    def get_nfo_filename(self, entry):
+        """
+        Get the filename of the nfo file from the 'location' in the entry.
+
+        Returns
+        -------
+        str
+            The file name of the 'nfo' file, or None it there is no 'nfo' file.
+        """
+        location = entry.get('location')
+        nfo_full_filename = os.path.splitext(location)[0] + self.nfo_file_extension
+
+        if os.path.isfile(nfo_full_filename):
+            return nfo_full_filename
+        else:
+            return None
+
+
+class BadXmlFile(Exception):
+    """
+    Exception that is raised if the nfo file can't be parsed due to some
+    invalid nfo file.
+    """
+    pass
+
+
+class NfoReader(object):
+    """
+    Class in charge of parsing the '.nfo' file and getting a dictionary of
+    fields.
+
+    The '.nfo' file is an XML file. Some fields can only appear once, such as
+    'title', 'id', 'plot', etc., while other fields can appear multiple times
+    (with different values), such as 'thumb', 'genre', etc. These fields are
+    listed in the `_fields` attribute.
+    """
+    def __init__(self, filename):
+        from xml.etree.ElementTree import ParseError
+
+        try:
+            tree = ET.parse(filename)
+            root = tree.getroot()
+        except ParseError:
+            raise BadXmlFile()
+
+        if os.path.exists(filename):
+            self._nfo_filename = filename
+            self._root = root
+        else:
+            raise BadXmlFile()
```

Comment:
Seems a bit silly to use the same exception for ParseError and if the file doesn't exist. At the very least you should give it a proper error message.